Summary:
[hbase] Move at end of compaction sometimes fails
Description:
I've seen the following exception a couple of times loading a cluster with data
2007-08-21 23:52:08,079 INFO org.apache.hadoop.hbase.HRegion: starting compaction on region hbaserepository,Rc0CnuSEIxXeO1CAcLTV2V==,221090350699527359
2007-08-21 23:52:08,079 DEBUG org.apache.hadoop.hbase.HStore: started compaction of 3 files in /bfd/hadoop-stack-data/tmp/hbase/compaction.tmp/hregion_hbaserepository,Rc0CnuSEIxXeO1CAcLTV2V==,221090350699527359/repo
2007-08-21 23:52:13,129 DEBUG org.apache.hadoop.hbase.HStore: moving hbaserepository,Rc0CnuSEIxXeO1CAcLTV2V==,221090350699527359/repo/-1 in /bfd/hadoop-stack-data/tmp/hbase/compaction.tmp to hbaserepository,Rc0CnuSEIxXeO1CAcLTV2V==,221090350699527359/repo/4863512705511503252 in /bfd/hadoop-stack-data/tmp/hbase
2007-08-21 23:52:13,214 ERROR org.apache.hadoop.hbase.HRegionServer: What happens if this fails? Are we toast?
org.apache.hadoop.ipc.RemoteException: java.io.IOException: File does not exist
        at org.apache.hadoop.dfs.FSDirectory.getFileInfo(FSDirectory.java:747)
        at org.apache.hadoop.dfs.FSNamesystem.getFileInfo(FSNamesystem.java:1253)
        at org.apache.hadoop.dfs.NameNode.getFileInfo(NameNode.java:459)
        at sun.reflect.GeneratedMethodAccessor19.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
        at java.lang.reflect.Method.invoke(Method.java:597)
        at org.apache.hadoop.ipc.RPC$Server.call(RPC.java:340)
        at org.apache.hadoop.ipc.Server$Handler.run(Server.java:566)

        at org.apache.hadoop.ipc.Client.call(Client.java:470)
        at org.apache.hadoop.ipc.RPC$Invoker.invoke(RPC.java:165)
        at org.apache.hadoop.dfs.$Proxy0.getFileInfo(Unknown Source)
        at sun.reflect.GeneratedMethodAccessor10.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
        at java.lang.reflect.Method.invoke(Method.java:597)
        at org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:82)
        at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:59)
        at org.apache.hadoop.dfs.$Proxy0.getFileInfo(Unknown Source)
        at org.apache.hadoop.dfs.DFSClient.getFileInfo(DFSClient.java:431)
        at org.apache.hadoop.dfs.DistributedFileSystem.getFileStatus(DistributedFileSystem.java:342)
        at org.apache.hadoop.fs.FileSystem.getLength(FileSystem.java:449)
        at org.apache.hadoop.io.SequenceFile$Reader.<init>(SequenceFile.java:1220)
        at org.apache.hadoop.io.SequenceFile$Reader.<init>(SequenceFile.java:1215)
        at org.apache.hadoop.io.MapFile$Reader.<init>(MapFile.java:228)
        at org.apache.hadoop.io.MapFile$Reader.<init>(MapFile.java:216)
        at org.apache.hadoop.hbase.HStoreFile$BloomFilterMapFile$Reader.<init>(HStoreFile.java:813)
        at org.apache.hadoop.hbase.HStoreFile.getReader(HStoreFile.java:925)
        at org.apache.hadoop.hbase.HStore.processReadyCompaction(HStore.java:996)
        at org.apache.hadoop.hbase.HStore.compactHelper(HStore.java:658)
        at org.apache.hadoop.hbase.HStore.compactHelper(HStore.java:564)
        at org.apache.hadoop.hbase.HStore.compact(HStore.java:559)
        at org.apache.hadoop.hbase.HRegion.compactStores(HRegion.java:700)
        at org.apache.hadoop.hbase.HRegionServer$SplitOrCompactChecker.run(HRegionServer.java:169)
        at java.lang.Thread.run(Thread.java:619)
Looking on the filesystem, I see the following for the problematic store:
/bfd/hadoop-stack-data/tmp/hbase/hregion_hbaserepository,Rc0CnuSEIxXeO1CAcLTV2V==,221090350699527359/repo/info  <dir>           2007-08-21 23:52
/bfd/hadoop-stack-data/tmp/hbase/hregion_hbaserepository,Rc0CnuSEIxXeO1CAcLTV2V==,221090350699527359/repo/info/4863512705511503252      <r 3>   9       2007-08-21 23:52
/bfd/hadoop-stack-data/tmp/hbase/hregion_hbaserepository,Rc0CnuSEIxXeO1CAcLTV2V==,221090350699527359/repo/mapfiles      <dir>           2007-08-21 23:52
/bfd/hadoop-stack-data/tmp/hbase/hregion_hbaserepository,Rc0CnuSEIxXeO1CAcLTV2V==,221090350699527359/repo/mapfiles/4863512705511503252  <dir>           2007-08-21 23:52
/bfd/hadoop-stack-data/tmp/hbase/hregion_hbaserepository,Rc0CnuSEIxXeO1CAcLTV2V==,221090350699527359/repo/mapfiles/4863512705511503252/index    <r 3>   8340    2007-08-21 23:52
Status:
CLOSED
Priority:
Major
Resolution:
Duplicate
Affects_version:
None
Fix_version:
None
Component:
None
Label:
None
Environment:

Attachment number:
0
Assignee:
Unassigned
Reporter:
stack
Create date:
22/Aug/07 00:11
Update date:
04/Feb/08 18:40
Resolved date:
24/Aug/07 19:53
